README.md

```diff
@@ -14,83 +16,179 @@ yarn add --dev eslint-config-yscope
+Vite project created from the `react-ts` template, you can use the helper function 
+`createTsConfigOverride` to generate configuration overrides. This ensures proper handling of
+imports and resolution.
+   ```javascript
+   import CommonConfig from "eslint-config-yscope/CommonConfig.mjs";
+   import ReactConfigArray from "eslint-config-yscope/ReactConfigArray.mjs";
+   import StylisticConfigArray from "eslint-config-yscope/StylisticConfigArray.mjs";
+   import TsConfigArray, {createTsConfigOverride} from "eslint-config-yscope/TsConfigArray.mjs";
+   
+   
+   const EslintConfig = [
+       {
+           ignores: [
+               "dist/",
+               "node_modules/",
+           ],
+       },
+       CommonConfig,
+       ...TsConfigArray.map(
+           (config) => ({
+               files: [
+                   "**/*.ts",
+                   "**/*.tsx",
+               ],
+               ...config,
+           })
+       ),
+       createTsConfigOverride(
+           [
+               "src/**/*.ts",
+               "src/**/*.tsx",
+           ],
+           "tsconfig.app.json"
+       ),
+       createTsConfigOverride(
+           ["vite.config.ts"],
+           "tsconfig.node.json"
+       ),
+       ...StylisticConfigArray,
+       ...ReactConfigArray,
+   ];
+   
+   
+   export default EslintConfig;
+   ```
 
-```json5
-{
-  "overrides": [
-    // ...
-    {
-      "files": ["test/**/*"],
-      "extends": ["yscope/jest"]
-    }
-  ]
-}
-```
-
-### For TypeScript Configuration
-
-TypeScript-related dependencies are not automatically installed because not every configuration requires them.
-Before you can use this configuration, you need to install the dependencies manually.
 
-```shell
-npm install --save-dev eslint-import-resolver-typescript @typescript-eslint/parser @typescript-eslint/eslint-plugin
-```
-
-Then in your ESLint configuration:
-
-```json
-{
-  "extends": [
-    "yscope/common",
-    "yscope/typescript"
-  ]
-}
-```
 
 ## Customization
 
-Each configuration can be further customized and extended as per your project's needs. You can override specific rules by adding them to your ESLint configuration file.
+Each configuration can be further customized and extended as per your project's needs. You can
+override specific rules by adding them to your ESLint configuration file.
 
 ## Contributing
 
-We welcome contributions to the `eslint-config-yscope`! If you have suggestions or improvements, feel free to open an issue or a pull request.
+We welcome contributions to the `eslint-config-yscope`! If you have suggestions or improvements,
+feel free to open an issue or a pull request.
 
 ## License
 
-This project is licensed under Apache License 2.0. Please see the [LICENSE](https://github.com/y-scope/eslint-config-yscope/blob/main/LICENSE) file for more details.
+This project is licensed under Apache License 2.0. Please see the [LICENSE][license] file for more
+details.
```

Comment:
Do we need this? GitHub already specifies the license, no? If you want to specify the license on npmjs.org, I assume we need to fill out the license field in package.json?

Reply:
yeah we don't have to add this into the README

i'll add the license field to the package.json